File: app/src/main/java/com/example/yelpbusinessexample/data/api/YelpServiceApi.java
package com.example.yelpbusinessexample.data.api;

import com.example.yelpbusinessexample.data.model.BusinessResponse;

import java.util.Map;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.Headers;
import retrofit2.http.QueryMap;

public interface YelpServiceApi {

    @GET("businesses/search")
    Call<BusinessResponse> getBusinessesByDistance(@Header("Authorization") String authorization, @QueryMap Map<String, String> searchQueries);

}

File: app/src/main/java/com/example/yelpbusinessexample/data/api/YelpServiceApi.kt
package com.example.yelpbusinessexample.data.api

import retrofit2.http.GET
import com.example.yelpbusinessexample.data.model.BusinessResponse
import retrofit2.Call
import retrofit2.http.Header
import retrofit2.http.QueryMap

interface YelpServiceApi {

    @GET("businesses/search")
    fun getBusinessesByDistance(
        @Header("Authorization") authorization: String,
        @QueryMap searchQueries: Map<String, String>
    ): Call<BusinessResponse?>
}